src/data_watcher.hpp

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <sys/inotify.h>
+#include <unistd.h>
+
+#include <sdbusplus/async.hpp>
+
+#include <filesystem>
+
+namespace data_sync
+{
+namespace watch::inotify
+{
+
+namespace fs = std::filesystem;
+
+/**
+ * @brief A tuple which has the info related to the occured inotify event
+ *
+ * std::string - name[] in inotify_event struct
+ * uint32_t    - Mask describing event
+ */
+using BaseName = std::string;
+using EventMask = uint32_t;
+using EventInfo = std::tuple<BaseName, EventMask>;
+
+/** @class FD
+ *
+ *  @brief RAII wrapper for file descriptor.
+ */
+class FD
+{
+  private:
```

Comment:
not a big deal, but better to be consistent on if public or private is first.  Usually public is.